### PR TITLE
Implementing feedback

### PR DESCRIPTION
**Pull request by tobygabriella**
- Reduce Api calls
- Implemented a utils class to reduce repeated code

### CODE REVIEW

File: Frontend/src/Components/DetailContainer/ListingDetailContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from "react";
```
Comment (conniewu3) [inline Frontend/src/Components/DetailContainer/ListingDetailContainer.jsx:1]: love this refactor!!!

File: Frontend/src/App.jsx [hunk 2]
```diff
@@ -52,22 +54,24 @@ function App() {
               <Route path="/" element={<Home />} />
               <Route path="/userProfile" element={<UserProfile />} />
               <Route path="/createListing" element={<CreateListing />} />
-              <Route path="/listings/:id" element={<ListingDetails />} />
+              <Route path="/listings/:id" element={<ListingDetailContainer />}>
+                <Route index element={<ListingDetails />} />
```
Comment (conniewu3) [inline Frontend/src/App.jsx:58]: nice use of an index route!

**Review by conniewu3 — APPROVED** [review]
looks good!

File: Frontend/src/Components/ListingItem/ListingItem.jsx [hunk 3]
```diff
@@ -4,41 +4,15 @@ import "./ListingItem.css";
...
-  }, [id]);
+const ListingItem = ({
+  id,
+  title,
+  price,
+  imageUrls,
+  status,
+  liked: initialLiked,
+}) => {
+  const [liked, setLiked] = useState(initialLiked);
```
Comment (conniewu3) [inline Frontend/src/Components/ListingItem/ListingItem.jsx:15]: nice use of initial state!

File: Frontend/src/Components/utils/likeStatusUtil.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const API_KEY = import.meta.env.VITE_BACKEND_ADDRESS;
+
+export const fetchListingsWithStatusAndLiked = async (listings) => {
+  const listingsWithStatusAndLiked = await Promise.all(
+    listings.map(async (listing) => {
+      const statusResponse = await fetch(`${API_KEY}/listings/${listing.id}`);
```
Comment (conniewu3) [inline Frontend/src/Components/utils/likeStatusUtil.js:6]: is status not already returned with listings from the fetch call?

File: Frontend/src/Components/utils/likeStatusUtil.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+const API_KEY = import.meta.env.VITE_BACKEND_ADDRESS;
+
+export const fetchListingsWithStatusAndLiked = async (listings) => {
```
Comment (conniewu3) [inline Frontend/src/Components/utils/likeStatusUtil.js:3]: is there any scenario when fetching listings that we don't want the status and liked data? if not, maybe we could do this logic on the back-end before returning listings in each get listings API call (the logic would be in a helper function as well)?

Comment (tobygabriella) [reply Frontend/src/Components/utils/likeStatusUtil.js:3]: i don't think there is a scenario where we do not require the status and liked data. All the places that call it, requires it

**Pull request merged**